### PR TITLE
Provide a way for MAS admins to delete an entire firm

### PR DESCRIPTION
**Pull request by munckymagik**
# What?

Adds a button to the private MAS admin UI that allows deletion of a firm and all related trading names from the directory.

**Note** Counter-intuitively this is done by deleting the principal as the principal object "owns" the parent firm and trading names so deleting that cascades through correctly.

# Why?

Whenever firms disappear from the FCA register or ask to be taken off of the directory we need a way to delete them. IIRC the frequency has been about 2 or 3 a month.

Until now, firm deletion has been a BAU task done on the Rails console whenever it was necessary.

# Screenshots

screen shot 2016-02-26 at 15 24 22

screen shot 2016-02-26 at 15 18 47

### CODE REVIEW

**Comment by munckymagik** [comment]
@rnalexander you may like this.

**Comment by Tr4pSt3R** [comment]
Woah this is a dangerous button!

File: app/controllers/admin/principals_controller.rb [hunk 1]
```diff
@@ -8,6 +8,19 @@ def show
     @principal = principal
   end
 
+  def destroy
```
Comment (Tr4pSt3R) [inline app/controllers/admin/principals_controller.rb]: Hey @munckymagik, Is there sufficient authorisation check to ensure that it is only the admin who can destroy a principle please?

Comment (munckymagik) [inline app/controllers/admin/principals_controller.rb]: Sure the entire `/admin` route is wrapped in HTTP auth:

https://github.com/moneyadviceservice/rad/blob/master/config/routes.rb#L81

screen shot 2016-02-26 at 15 54 15

Comment (Tr4pSt3R) [inline app/controllers/admin/principals_controller.rb]: @munckymagik that looks good

**Comment by Tr4pSt3R** [comment]
:+1:

**Comment by cazrin** [comment]
:+1:

**Comment by munckymagik** [comment]
Thanks for the review @Tr4pSt3R @cazrin

**Pull request merged**